README.md

```diff
@@ -160,7 +214,99 @@ Deploy the application to Minikube using the Marblerun.
+    ```
+
+    We can now update the image used by the emojivoto voting statefulset:
+
+    ```bash
+    kubectl set image -n emojivoto statefulset/voting voting-svc=ghcr.io/edgelesssys/emojivoto/voting-svc:v0.3.0-fix
+    ```
+
+    Wait for the pod to restart. Once it is up and running again you can reconnect to the webpage and vote for the Doughnut as your favorite emoji!
+
+    >It might be neccessary to restart some components of the application. If you receive any errors when trying to vote you can rollout a restart and then reconnect to the webpage:
+    >```bash
+    >kubectl rollout restart -n emojivoto statefulset emoji web
+    >```
+
+1. Bringing the Coordinator into recovery mode
+
+    As already mentioned, some cases require manual intervention to recover the Coordinator, for example if the host, your virtual machine was running on, changes.
+    
+    On minikube we can simulate this behavior by first stopping minikube, and then deleting the `sealed_key` of the Coordinator.
+
+    * If you are running minikube in docker, `sealed_key` can be removed as follows:
+    
+    ```bash
+    minikube stop
+    rm /var/lib/docker/volumes/minikube/_data/hostpath-provisioner/marblerun/coordinator-pv-claim/sealed_key
+    ```
+
+    * Restart minikube
+
+    ```bash
+    minikube start
+    ```
+
+    * Note that the previously running emojivoto pods will now fail to start. Check the status of the coordinator:
+
+    ```bash
+    marblerun status $MARBLERUN [--insecure]
+    ```
+
+    * The output should be the following:
+
+    ```bash
+    1: Coordinator is in recovery mode. Either upload a key to unseal the saved state, or set a new manifest. For more information on how to proceed, consult the documentation.
+    ```
+
+1. Decrypting the Recovery Key
+
+    Luckily we provided a recovery key when we first set the manifest. We can now decrypt the recovery secret we received from the coordinator:
+
+    ```bash
+    cat recovery.json | jq -r '.RecoverySecrets.recoveryKey1' | base64 -d > recovery_key_encrypted
+    openssl pkeyutl -inkey recovery_priv.key -in recovery_key_encrypted -pkeyopt rsa_padding_mode:oaep -pkeyopt rsa_oaep_md:sha256 -decrypt -out recovery_key_decrypted
+    ```
+
+1. Recovering the Coordinator
+
+    Now we can upload the key and recover Marblerun without losing data:
+
+    ```bash
+    marblerun recover $MARBLERUN recovery_key_decrypted [--insecure]
+    ```
+
+    If the recovery was successful all emojivoto pods can once again start correctly.
+
+1. Scaling your application
+
+    Increasing or decreasing the number of pods running in your application requires no extra steps with Marblerun.
+
+    * If you want to scale the web frontend to use 3 pods:
+
+    ```bash
+    kubectl scale statefulset -n emojivoto web --replicas=3
+    ```
+
+    * And scale it back down to only 1:
+
+    ```bash
+    kubectl scale statefulset -n emojivoto web --replicas=1
```

Comment:
```suggestion
1. (Optional) Update emojivoto

    We would like to demonstrate an update procedure with Marblerun.
    If you have voted for the Doughnut emoji you have been presented with an error message. This is an intentional "bug" in the application that we are going to fix now by rolling out a patched emojivoto version.

    In a non-confidential environment, you could simply replace the currently running version with the patched one. In our case, the Coordinator prevents unauthorized alterations to the deployment and only accepts versions as previously specified in the manifest.

    To avoid having to redeploy your entire application from scratch, Marblerun allows its users to upload a separate "Update Manifest" which defines new versions of a previously set `Package`.

    To upload the "Update Manifest" we need to authenticate ourselves to the Coordinator using the previously created admin key and certificate:

    ```bash
    marblerun manifest update tools/update-manifest.json $MARBLERUN --cert admin_certificate.crt --key admin_private.key [--insecure]
    ```

    We can now update the image used by the emojivoto voting Statefulset:

    ```bash
    kubectl set image -n emojivoto statefulset/voting voting-svc=ghcr.io/edgelesssys/emojivoto/voting-svc:v0.3.0-fix
    ```
    Updating the manifest will invalidate Marblerun's certificate chain so that the existing services will not accept old versions of the updated voting service anymore. Hence, we need to restart the other services to obtain a fresh certificate chain:
    >```bash
    > kubectl rollout restart -n emojivoto statefulset emoji emoji
    > kubectl rollout restart -n emojivoto statefulset emoji web
    >```

    Wait for the pods to restart. Once they are up and running again you can reconnect to the webpage and vote for the Doughnut as your favorite emoji!



1. (Optional) Bringing the Coordinator into recovery mode

    As already mentioned, some cases require manual intervention to recover the Coordinator, for example, if the host, the Coordinator was running on, changes.
    
    On minikube, we can simulate this behavior by first stopping minikube, and then deleting the `sealed_key` of the Coordinator.

    * If you are running minikube in docker, `sealed_key` can be removed as follows:
    
    ```bash
    minikube stop
    rm /var/lib/docker/volumes/minikube/_data/hostpath-provisioner/marblerun/coordinator-pv-claim/sealed_key
    ```

    * Restart minikube

    ```bash
    minikube start
    ```

    * Note that the previously running emojivoto pods will now fail to start. Check the status of the coordinator:

    ```bash
    marblerun status $MARBLERUN [--insecure]
    ```

    * The output should be the following:

    ```bash
    1: Coordinator is in recovery mode. Either upload a key to unseal the saved state, or set a new manifest. For more information on how to proceed, consult the documentation.
    ```

1. (Optional) Decrypting the Recovery Key

    Luckily we provided a recovery key when we first set the manifest. We can now decrypt the recovery secret we received from the coordinator:

    ```bash
    cat recovery.json | jq -r '.RecoverySecrets.recoveryKey1' | base64 -d > recovery_key_encrypted
    openssl pkeyutl -inkey recovery_priv.key -in recovery_key_encrypted -pkeyopt rsa_padding_mode:oaep -pkeyopt rsa_oaep_md:sha256 -decrypt -out recovery_key_decrypted
    ```

1. (Optional) Recovering the Coordinator

    Now we can upload the key and recover the Marblerun state:

    ```bash
    marblerun recover $MARBLERUN recovery_key_decrypted [--insecure]
    ```

    If the recovery was successful all emojivoto pods can once again start correctly.

1. (Optional) Scaling your application

    Increasing or decreasing the number of pods running in your application requires no extra steps with Marblerun.

    * If you want to scale the web frontend to use 3 pods:

    ```bash
    kubectl scale statefulset -n emojivoto web --replicas=3
    ```

    * And scale it back down to only 1:

    ```bash
    kubectl scale statefulset -n emojivoto web --replicas=1
```

Please make the recovery process one step with multiple substeps.